services/sonar/sonar-quality-gate.service.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const SonarBase = require('./sonar-base')
+const { patternBase, queryParamSchema } = require('./sonar-helpers')
+
+module.exports = class SonarQualityGate extends SonarBase {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'quality gate' }
+  }
+
+  static render({ qualityState }) {
+    if (qualityState === 'OK') {
+      return {
+        message: 'passed',
+        color: 'success',
+      }
+    }
+    return {
+      message: 'failed',
+      color: 'critical',
+    }
+  }
+
+  static get route() {
+    return {
+      base: 'sonar',
+      pattern: `${patternBase}/:metric(quality_gate|alert_status)`,
```

Comment:
The metric sonar uses under the hood is `alert_status` but in all the UI/reporting this is presented to users as Quality Gate. I added support for both in the route as I think `quality_gate` will make more sense to the average user, but also kept `alert_status` since some users may use that instead